pkg/chunk/cached_store.go

```diff
@@ -1099,8 +1101,8 @@ func (store *cachedStore) NewReader(id uint64, length int) Reader {
 	return sliceForRead(id, length, store)
 }
 
-func (store *cachedStore) NewWriter(id uint64) Writer {
-	return sliceForWrite(id, store)
+func (store *cachedStore) NewWriter(id uint64, forceUpload bool) Writer {
```

Comment:
Don't change this function. Instead add a new one `SetWriteback(enabled bool)` to adjust the configuration.